@navikt/core/react/src/util/focus-boundary/FocusBoundary.tsx

```diff
@@ -166,88 +175,129 @@ const FocusBoundary = forwardRef<HTMLDivElement, FocusBoundaryProps>(
-        initialFocusedElement && container.contains(initialFocusedElement);
-
-      /*
-       * We only autofocus on mount if container does not contain active element.
-       * If container has an element with `autoFocus` attribute, browser will
-       * have already moved focus there before this effect runs.
-       */
-      if (!containsActiveElement) {
-        const mountEvent = new CustomEvent(AUTOFOCUS_ON_MOUNT, EVENT_OPTIONS);
-        container.addEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus);
-        container.dispatchEvent(mountEvent);
 
-        /* If consumer does not manually prevent event and handle focus themselves */
-        if (!mountEvent.defaultPrevented) {
-          /**
-           * Attempts focusing the first element in a list of candidates.
-           * Stops when focus has actually moved.
-           */
-          const candidates = removeLinks(getTabbableCandidates(container));
-          const previouslyFocusedElement = document.activeElement;
-          for (const candidate of candidates) {
-            focus(candidate, { select: true });
-            if (document.activeElement !== previouslyFocusedElement) {
-              break;
-            }
-          }
+      return () => {
+        setTimeout(() => {
+          focusBoundarysStack.remove(focusBoundary);
+        }, 0);
+      };
+    }, [container, focusBoundary]);
 
-          /* focusFirst might not find any candidates, so we fall back to focusing container */
-          if (document.activeElement === initialFocusedElement) {
-            focus(container);
-          }
+    useEffect(() => {
+      if (!container || !modal) {
+        return;
+      }
+
+      return hideNonTargetElements([container]);
+    }, [container, modal]);
+
+    /* Handles mount focus */
+    useClientLayoutEffect(() => {
+      if (!container || initialFocusRef.current === false) {
+        return;
+      }
+
+      const ownerDoc = ownerDocument(container);
+      const previouslyFocusedElement = ownerDoc.activeElement;
+
+      queueMicrotask(() => {
+        const focusableElements = removeLinks(getTabbableCandidates(container));
+        const initialFocusValueOrFn = initialFocusRef.current;
+        const resolvedInitialFocus =
+          typeof initialFocusValueOrFn === "function"
+            ? initialFocusValueOrFn()
+            : initialFocusValueOrFn;
+
+        if (
+          resolvedInitialFocus === undefined ||
+          resolvedInitialFocus === false
+        ) {
+          return;
+        }
+
+        let elToFocus: HTMLElement | null | undefined;
+
+        /* `null` should fallback to default behavior in case of an empty ref. */
+        if (resolvedInitialFocus === true || resolvedInitialFocus === null) {
+          elToFocus = focusableElements[0] || container;
+        } else {
+          elToFocus =
+            resolveRef(resolvedInitialFocus) ||
+            focusableElements[0] ||
+            container;
+        }
```

Comment:
I think we can just do this now😅

```suggestion
        const elToFocus =
          (resolvedInitialFocus !== true &&
            resolvedInitialFocus &&
            resolveRef(resolvedInitialFocus)) ||
          focusableElements[0] ||
          container;
```

Reply:
While this simplifies it, i like the clarity the if-statement gives now. But made a small change 
```
        let elToFocus: HTMLElement | null | undefined;
        const fallbackelements = focusableElements[0] || container;

        /* `null` should fallback to default behavior in case of an empty ref. */
        if (resolvedInitialFocus === true || resolvedInitialFocus === null) {
          elToFocus = fallbackelements;
        } else {
          elToFocus = resolveRef(resolvedInitialFocus) || fallbackelements;
        }
```